### PR TITLE
address chef 14.3.x shell_out deprecations

### LINKED ISSUE #109: undefined method `clean_array' error on `hab_package` invocation
### Cookbook version
0.57.0

### Chef-client version
14.3.37

### Platform Details
Ubuntu 16.04 (Azure)

### Scenario:
Simple installation of hab packaged national parks (https://github.com/chef-partners/hab_national_parks)

### Steps to Reproduce:
Using Chef run on checked out version of cookbook e.g. 

```
chef-run  hab-test.westus2.cloudapp.azure.com hab_national_parks   --sudo --user james
```

### Expected Result:
Supervisor and packages installed

### Actual Result:
```
 chef-run  hab-test.westus2.cloudapp.azure.com hab_national_parks   --sudo --user james
[\] Generating local policyfile...WARN: scripts cookbook missing metadata or no name field, skipping
[✔] Generated local policyfile
[✔] [hab-test.westus2.cloudapp.azure.com] Connected.
[✔] [hab-test.westus2.cloudapp.azure.com] Chef client version 14.3.37 already installed on target.
[✖] [hab-test.westus2.cloudapp.azure.com] Failed to converge target.

CHEFCCR002

The converge of the remote host failed for the
following reason:

  undefined method `clean_array' for #<Chef::Provider::Package::Hart:0x0000000003e91b70>

If you are not able to resolve this issue, please contact Chef support
at beta@chef.io

```

Logs:
```
Synchronizing Cookbooks:
  - hab_national_parks (0.1.1)
  - habitat (0.57.0)
Installing Cookbook Gems:
Compiling Cookbooks...
Converging 4 resources
Recipe: hab_national_parks::default
  * hab_install[install habitat] action install
    * apt_package[curl, tar] action install (up to date)
    * linux_user[hab] action create (up to date)
     (up to date)
  * hab_sup_systemd[default] action run
    * hab_install[default] action install
      * apt_package[curl, tar] action install (up to date)
      * linux_user[hab] action create (up to date)
       (up to date)
    * hab_package[core/hab-sup] action install

      ================================================================================
      Error executing action `install` on resource 'hab_package[core/hab-sup]'
      ================================================================================

      NoMethodError
      -------------
      undefined method `clean_array' for #<Chef::Provider::Package::Hart:0x0000000003e91b70>

      Cookbook Trace:
      ---------------
      /var/chef-workstation/cache/cookbooks/habitat/libraries/provider_hab_package.rb:90:in `hab'
      /var/chef-workstation/cache/cookbooks/habitat/libraries/provider_hab_package.rb:144:in `installed_version'
      /var/chef-workstation/cache/cookbooks/habitat/libraries/provider_hab_package.rb:139:in `block in current_versions'
      /var/chef-workstation/cache/cookbooks/habitat/libraries/provider_hab_package.rb:138:in `map'
      /var/chef-workstation/cache/cookbooks/habitat/libraries/provider_hab_package.rb:138:in `current_versions'
      /var/chef-workstation/cache/cookbooks/habitat/libraries/provider_hab_package.rb:59:in `load_current_resource'

      Resource Declaration:
      ---------------------
      # In /var/chef-workstation/cache/cookbooks/habitat/libraries/resource_hab_sup.rb

       43:         hab_package 'core/hab-sup' do
       44:           bldr_url new_resource.bldr_url if new_resource.bldr_url
       45:         end
       46:       end


```

### PR DESCRIPTION
**Pull request by lamont-granquist**
shell_out_with_timeout + clean_array are baked into the shell_out
API now post-14.3.20

closes #109

### CODE REVIEW

**Comment by jamesc** [comment]
+1  This works with Chef 14.3.37

**Comment by lamont-granquist** [comment]
@jtimberman @tas50 or someone wanna merge + release?

**Comment by tas50** [comment]
We should configure kitchen to ignore that deprecation for now otherwise travis will be busted

**Comment by lamont-granquist** [comment]
something is busted then, because this was supposed to fix that...

File: resources/config.rb [hunk 1]
```diff
@@ -53,13 +53,17 @@
     opts << ['--org', new_resource.org] if new_resource.org
     opts << ['--ring', new_resource.ring] if new_resource.ring
 
+    command = [ 'hab', 'config', 'apply', opts, new_resource.service_group,
```
Comment (jamesc) [inline resources/config.rb]: `tempfile` not created yet - need to move this down two lines

**Pull request merged**